### ISSUE TITLE
How can l use the weights trained by GreenMIM as the pre-trained weights for official swing transformer?

### ISSUE THREAD

**Issue by Geek-Peng** [LayneH/GreenMIM#4]
Hi ,Thanks for the great work!
I'v tried GreenMIM on my data and it really worked well(I have visualized the masked patch generated by the model). And then i want to use this weights as official swing transformer classification work pre-trained weights but i find these weights are not in  GreenMIM :
   'head.weight', 'head.bias', 'layers.0.blocks.1.attn_mask', 'layers.1.blocks.1.attn_mask', 'layers.2.blocks.1.attn_mask', 
   'layers.2.blocks.3.attn_mask', 'layers.2.blocks.5.attn_mask', 'layers.2.blocks.7.attn_mask', 'layers.2.blocks.9.attn_mask', 
   'layers.2.blocks.11.attn_mask', 'layers.2.blocks.13.attn_mask', 'layers.2.blocks.15.attn_mask', 'layers.2.blocks.17.attn_mask'
I've simply use official swing transformer init code as follow:
    def _init_weights(self, m):
        if isinstance(m, nn.Linear):
            trunc_normal_(m.weight, std=.02)
            if isinstance(m, nn.Linear) and m.bias is not None:
                nn.init.constant_(m.bias, 0)
        elif isinstance(m, nn.LayerNorm):
            nn.init.constant_(m.bias, 0)
            nn.init.constant_(m.weight, 1.0)
Are there any better ways? Many thanks! I would be highly appreciated.

**Comment by LayneH** [comment]
Hi,

The logging info is as expected because we do not have the `head` (i.e., a 1000-way classifier) at the pre-training and the `attn_mask` will be generated automatically as in [these few lines](https://github.com/microsoft/SimMIM/blob/main/models/swin_transformer.py#L209).
Thus it is safe to directly use the official fine-tuning code of Swin Transformer without any change, as your provided code snippet.

**Comment by Geek-Peng** [comment]
> Hi,
> 
> The logging info is as expected because we do not have the `head` (i.e., a 1000-way classifier) at the pre-training and the `attn_mask` will be generated automatically as in [these few lines](https://github.com/microsoft/SimMIM/blob/main/models/swin_transformer.py#L209). Thus it is safe to directly use the official fine-tuning code of Swin Transformer without any change, as your provided code snippet.

OK!Thank you so much!
